File: School_Food_Android/data/src/main/java/com/meals/data/network/remote/MealRemote.kt
package com.meals.data.network.remote

import com.meals.data.network.service.MealService
import io.reactivex.Single
import com.meals.data.base.BaseRemote
import com.meals.data.network.response.MealData

class MealRemote(override val service: MealService): BaseRemote<MealService>() {

    fun getMeal(id: String, code: String, date: String): Single<List<String?>> =
        service.getMeals(id, code, date).map(getResponseData()).map(MealData::meals)
}

File: School_Food_iOS_V2/School_Food_iOS_V2/Data/Network/Remote/MealRemote.swift
//
//  MealRemote.swift
//  School_Food_iOS_V2
//
//  Created by 박진 on 2020/12/01.
//

import Foundation
import Moya
import RxSwift
import RxCocoa

class MealRemote {
    let provider: MoyaProvider<MealAPI>
    
    init() {
        provider = MoyaProvider()
    }
    
    func getMeal(schoolId: String, officeCode: String, date: String) -> Single<Array<String>> {
        provider.rx.request(.getMeal(schoolId: schoolId, officeCode: officeCode, date: date))
            .map(Response<MealData>.self, using: JSONDecoder())
            .map { response -> Array<String> in
                return response.data.meals
            }
    }
}
